Jenkinsfile

```diff
@@ -1,38 +1,31 @@
 pipeline {
-    agent none
+    agent { label 'dockerhost' }
     environment {
         CI = 'true'
+        DOCKER_REPO_NAME = "kiva/kiosa-dev"
         TAG_NAME = "${env.BRANCH_NAME}-${env.BUILD_NUMBER}"
+        TAGGED_IMAGE_NAME = "${env.DOCKER_REPO_NAME}:${env.TAG_NAME}"
     }
     stages {
-        stage('Test') {
-            agent {
-                dockerfile {
-                    args '-v $HOME/.gradle:/root/.gradle'
-                }
-            }
+        stage('Build and test') {
             steps {
-                echo 'Testing...'
-                sh "./gradlew build"
+                withDockerContainer(image: 'openjdk:8-jdk-alpine', args: '-v $HOME/.gradle:/root/.gradle') {
```

Comment:
We could do this with a docker agent declaration, but using this lets us maintain the dockerhost agent throughout the pipeline